staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go

```diff
@@ -0,0 +1,418 @@
+	// +optional
+	Status PriorityLevelConfigurationStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PriorityLevelConfigurationList is a list of PriorityLevelConfiguration objects.
+type PriorityLevelConfigurationList struct {
+	metav1.TypeMeta `json:",inline"`
+	// `metadata` is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// `items` is a list of request-priorities.
+	Items []PriorityLevelConfiguration `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// PriorityLevelConfigurationSpec is specification of a priority level
+type PriorityLevelConfigurationSpec struct {
+	// `queuingType` indicates whether this priority level does
+	// queuing or is exempt.  Valid values are "Queuing" and "Exempt".
+	// "Exempt" means that requests of this priority level are not subject
+	// to concurrency limits (and thus are never queued) and do not detract
+	// from the concurrency available for non-exempt requests. The "Exempt"
+	// type is useful for apiserver self-requests and system administrator use.
+	// Required.
+	Type PriorityLevelQueueingType `json:"type" protobuf:"varint,1,opt,name=queuingType"`
+
+	// `queuing` holds the configuration parameters that are
+	// only meaningful for a priority level that does queuing (i.e.,
+	// is not exempt).  This field must be non-empty if and only if
+	// `queuingType` is `"Queuing"`.
+	// +optional
+	Queuing *QueuingConfiguration `json:"queuing,omitempty" protobuf:"bytes,2,opt,name=queuing"`
+}
+
+// PriorityLevelQueueingType identifies the queuing nature of a priority level
+type PriorityLevelQueueingType string
+
+// Supported queuing types.
+const (
+	// PriorityLevelQueuingTypeQueueing is the PriorityLevelQueueingType for priority levels that queue
+	PriorityLevelQueuingTypeQueueing PriorityLevelQueueingType = "Queuing"
+
+	// PriorityLevelQueuingTypeExempt is the PriorityLevelQueueingType for priority levels that are exempt from concurrency controls
+	PriorityLevelQueuingTypeExempt PriorityLevelQueueingType = "Exempt"
+)
+
+// QueuingConfiguration holds the configuration parameters that are specific to a priority level that is subject to concurrency controls
+type QueuingConfiguration struct {
+	// `assuredConcurrencyShares` (ACS) must be a positive number. The
+	// server's concurrency limit (SCL) is divided among the
+	// concurrency-controlled priority levels in proportion to their
+	// assured concurrency shares. This produces the assured
+	// concurrency value (ACV) for each such priority level:
+	//
+	//             ACV(l) = ceil( SCL * ACS(l) / ( sum[priority levels k] ACS(k) ) )
+	//
+	// bigger numbers of ACS mean more reserved concurrent requests (at the
```

Comment:
Note that everywhere else we say "assured" rather than "reserved".

Reply:
Although reserved is more accurate, no? At least to me, reserved implies I have an exclusive claim while assured just means I won't be denied if I try to use it.

I think it is fine to continue to use the word assured, since I may be unique in that interpretation and we may change the behavior. But we could change this line to be "Larger values of ACS mean more assured (think: reserved) concurrent requests " since otherwise this isn't really explaining anything, it's just repeating the A in ACS.

Reply:
I think that for our first release, "reserved" is a better word than "assured".  I think we are using "assured" because we started with more ambitious ideas, for which "assured" is the better word.

I do not have a lot of enthusiasm for doing a comprehensive s/assured/reserved/, both because it is a lot of work and because we have ambitions that may lead us to wanting to reverse that reversal.

If you think the word "assured" needs some explanation, let's do that in its own sentence or more rather than sliding it into one parenthetic remark.

Reply:
> "assured" rather than "reserved"

(it's hard for me to figure out the differences between the two tho 🙂) keeping this open until we agree on which word wins..

Reply:
We're not going to make progress on that until we get input from more than @lavalamp and me.

Reply:
I vote we call it good on this comment and clarify more once we see more specifically how people are confused by it :)

Reply:
Right now the commentary uses mixed terminology, so that's certain to need fixing.  But we can fix the comments in later PRs.